src/main/scala/beam/agentsim/scheduler/BeamAgentScheduler.scala

```diff
@@ -338,6 +338,14 @@ class BeamAgentScheduler(
         } else {
           log.debug(s"Statistics about trigger: ${System.lineSeparator()} ${triggerMeasurer.getStat}")
         }
+        if (beamConfig.beam.debug.triggerMeasurer.enabled && beamConfig.beam.debug.triggerMeasurer.writeStuckAgentDetectionConfig) {
```

Comment:
Shouldn't this be run in the `TriggerMeasurer` itself? It seems like an unnecessary redirection

Reply:
In fact, I'm unsure why this is even in the measurer itself? Shouldn't it be in something else entirely?

Reply:
Originally I put it as a extra step in `getStat` method, but then I thought it's too much things for the method with the name `getStat`. Maybe I can call it `onIterationEnd` or something like that and leave all these logic there? @JustinPihony, what do you think?

Reply:
I see the dilemma...hard choice. Since you already made a choice this way I'd say go with it then :)